yt_dlp/extractor/txxx.py

```diff
@@ -0,0 +1,307 @@
+            'age_limit': 18,
+        }
+    }, {
+        'url': 'https://tubepornclassic.com/videos/1015455/mimi-rogers-full-body-massage-nude-compilation/',
+        'md5': '2e9a6cf610c9862e86e0ce24f08f4427',
+        'info_dict': {
+            'id': '1015455',
+            'ext': 'mp4',
+            'title': 'Mimi Rogers - Full Body Massage (Nude) compilation',
+            'uploader': '88bhuto',
+            'duration': 286,
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'age_limit': 18,
+        }
+    }, {
+        'url': 'https://upornia.com/videos/1498858/twistys-malena-morgan-starring-at-dr-morgan-baller/',
+        'md5': '7ff7033340bc88a173198b7c22600e4f',
+        'info_dict': {
+            'id': '1498858',
+            'ext': 'mp4',
+            'title': 'Twistys - Malena Morgan starring at Dr. Morgan-Baller',
+            'uploader': 'mindgeek',
+            'duration': 480,
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'age_limit': 18,
+        }
+    }, {
+        'url': 'https://vjav.com/videos/11761/yui-hatano-in-if-yui-was-my-girlfriend2/',
+        'md5': '6de5bc1f13bdfc3491a77f23edb1676f',
+        'info_dict': {
+            'id': '11761',
+            'ext': 'mp4',
+            'title': 'Yui Hatano in If Yui Was My Girlfriend',
+            'uploader': 'Matheus69',
+            'duration': 3310,
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'age_limit': 18,
+        }
+    }, {
+        'url': 'https://voyeurhit.com/videos/332875/charlotte-stokely-elle-alexandra-malena-morgan-lingerie/',
+        'md5': '12b4666e9c3e60dafe9182e5d12aae33',
+        'info_dict': {
+            'id': '332875',
+            'ext': 'mp4',
+            'title': 'Charlotte Stokely, Elle Alexandra, Malena Morgan-Lingerie',
+            'uploader': 'Kyle Roberts',
+            'duration': 655,
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'age_limit': 18,
+        }
+    }, {
+        'url': 'https://voyeurhit.tube/videos/332875/charlotte-stokely-elle-alexandra-malena-morgan-lingerie/',
+        'md5': '12b4666e9c3e60dafe9182e5d12aae33',
+        'info_dict': {
+            'id': '332875',
+            'ext': 'mp4',
+            'title': 'Charlotte Stokely, Elle Alexandra, Malena Morgan-Lingerie',
+            'uploader': 'Kyle Roberts',
+            'duration': 655,
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'age_limit': 18,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        group_1 = 0 if not video_id[:-3] else video_id[:-3] + '000'
+        group_2 = 0 if not video_id[:-6] else video_id[:-6] + '000000'
+        host = self._match_valid_url(url).group('host')
```

Comment:
```suggestion
        video_id, host = self._match_valid_url(url).group('id', 'host')
        group_1 = str(1E3 * (int(video_id) // 1E3))
        group_2 = str(1E6 * (int(video_id) // 1E6))
```

Reply:
Done.
The exponential notation returns a float and results in the string ending with ".0", so I used 1000 and 1000000 instead.